### PR TITLE
Add fix

### PR DESCRIPTION
**Pull request by brinxmat**
- Unfortunately, something went very wrong in our process here — let's not repeat this, tips about what went wrong?
- Added tests to describe what is expected
- Added refactored code to match

### CODE REVIEW

**Comment by mattiahj** [comment]
Both Master and Develop seem to build fine, so maybe the error is here?

File: doi-proxy/src/test/java/no/unit/nva/doi/CrossRefClientTest.java [hunk 1]
```diff
@@ -117,18 +135,82 @@ public void fetchDataForDoiReturnAnEmptyOptionalForANonExistingUrl()
...
 
+    @Test
+    void crossrefClientThrowsRuntimeExceptionIfEnvironmentVariableCrossrefApiTokenNameIsMissing() {
+        Environment mockEnvironment = mock(Environment.class);
+        when(mockEnvironment.readEnvOpt(CROSSREFPLUSAPITOKEN_KEY_ENV)).thenReturn(Optional.of("anything"));
+        Executable executable = () -> new CrossRefClient(HttpClient.newHttpClient(), mockEnvironment,
+                new SecretsReader());
+        Exception exception = assertThrows(RuntimeException.class, executable);
+        String actual = exception.getMessage();
+        assertThat(actual, containsString(MISSING_CROSSREF_TOKENS_ERROR_MESSAGE));
```
Comment (mattiahj) [inline doi-proxy/src/test/java/no/unit/nva/doi/CrossRefClientTest.java:158]: This does not assert what the name of the test implies it should assert, does it?

File: doi-proxy/src/main/java/no/unit/nva/doi/CrossRefClient.java [hunk 2]
```diff
@@ -102,11 +123,11 @@ private HttpRequest createRequest(URI doiUri) {
                 .header(HttpHeaders.USER_AGENT, CROSSREF_USER_AGENT)
                 .timeout(Duration.ofSeconds(TIMEOUT_DURATION))
                 .GET();
-        if (getCrossRefApiPlusToken().isPresent()) {
-            logger.info(ADDING_TOKEN_IN_HEADER);
-            builder.setHeader(CROSSREF_PLUSAPI_HEADER,
-                    String.format(CROSSREF_PLUSAPI_AUTHORZATION_HEADER_BASE, getCrossRefApiPlusToken().get()));
-        }
+
```
Comment (brinxmat) [inline doi-proxy/src/main/java/no/unit/nva/doi/CrossRefClient.java:126]: Before I refactor this, the previous functionality viewed misconfiguration of the secrets as OK and defaulted to the open Crossref API if no secret was found…do we still want this?

**Review by axthosarouris — CHANGES_REQUESTED** [review]
Temporary block because I want to check if the code can be made simpler. I will check it tomorrow first thing in the morning

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**